includes/Pages/PageViewRequest.php

```diff
@@ -55,12 +56,18 @@ protected function main()
         $domain = Domain::getById(1, $this->getDatabase());
         $this->assign('mediawikiScriptPath', $domain->getWikiArticlePath());
 
-        // Test we should be able to look at this request
-        if ($config->getEmailConfirmationEnabled()) {
-            if ($request->getEmailConfirm() !== 'Confirmed') {
-                // Not allowed to look at this yet.
-                throw new ApplicationLogicException('The email address has not yet been confirmed for this request.');
-            }
+        // Shows a page if the email is not confirmed.
+        if ($request->getEmailConfirm() !== 'Confirmed') {
+            // Show a banner if the user can manually confirm the request
+            $viewConfirm = $this->barrierTest("main", $currentUser, PageManuallyConfirm::class);
```

Comment:
Nit: Probably better to use the constant here?

```suggestion
            $viewConfirm = $this->barrierTest(RoleConfiguration::MAIN, $currentUser, PageManuallyConfirm::class);
```